src/App.tsx

```diff
@@ -14,11 +17,100 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isReversed: boolean,
+  Goods: string[],
+  listHidden: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    Goods: goodsFromServer,
```

Comment:
Use copy instead, cause in this case you assign `goodsFromServer` reference to `Goods` property. It can produce bugs
```suggestion
    Goods: [...goodsFromServer],
```